src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -137,29 +145,93 @@ def awsFilterImpairedNodes(nodes, ec2):
 class InvalidClusterStateException(Exception):
     pass
 
+class AWSConnectionManager:
```

Comment:
I really like this.  I'll have to figure out how to work it into some of the simpledb refactor PR.  One thing worth noting is that s3 claims that you have to start a new session each time in order to be thread-safe, so if we're parallelizing our downloads, we should probably account for that:

https://boto3.amazonaws.com/v1/documentation/api/1.14.31/guide/session.html#multithreading-or-multiprocessing-with-sessions